clients/src/main/java/io/strimzi/configuration/http/HttpClientsConfiguration.java

```diff
@@ -25,6 +27,11 @@ public HttpClientsConfiguration(Map<String, String> map) {
         long delay = ClientsConfigurationUtils.parseLongOrDefault(map.get(ConfigurationConstants.DELAY_MS_ENV), ConfigurationConstants.DEFAULT_DELAY_MS);
         int messageCount = ClientsConfigurationUtils.parseIntOrDefault(map.get(ConfigurationConstants.MESSAGE_COUNT_ENV), ConfigurationConstants.DEFAULT_MESSAGES_COUNT);
         String endpointPrefix = ClientsConfigurationUtils.parseStringOrDefault(map.get(ConfigurationConstants.ENDPOINT_PREFIX_ENV), ConfigurationConstants.DEFAULT_ENDPOINT_PREFIX);
+        this.messageType = ClientsConfigurationUtils.parseStringOrDefault(map.get(ConfigurationConstants.MESSAGE_TYPE_ENV), ConfigurationConstants.DEFAULT_MESSAGE_TYPE);
+
+        if (MessageType.getFromString(this.messageType) == MessageType.UNKNOWN) {
+            throw new InvalidParameterException("MESSAGE_TYPE should be 'json' or 'text'");
```

Comment:
Maybe suggestion -> you can create a small method in the `MessageType` to return list of the supported types (so in case that we will add a new one, we will not have to update this one as well).

Reply:
@Frawless still applies for both places?

Reply:
ye, forgot to update it, I will test it and post the fix later

Reply:
Updated